### ISSUE TITLE
Investigate slow compilation of Message module

### ISSUE THREAD

**Issue by sullyj3** [sullyj3/buttplug-hs-core#4]
I'm assuming the issue is deriving generic, might be worth switching to TH

**Comment by sullyj3** [comment]
Nice! I thought maybe an easy path might be to just use TH as a one off, view the result with `-ddump-splices`, and just commit the result. But unfortunately the result is horrible

```haskell
instance ToJSON Vibrate where
  toJSON
    = \ value_ay6F
        -> case value_ay6F of {
             Vibrate arg1_ay6G arg2_ay6H
               -> aeson-1.5.6.0:Data.Aeson.Types.ToJSON.fromPairs
                    ((aeson-1.5.6.0:Data.Aeson.Types.ToJSON.pair "Index")
                       (toJSON arg1_ay6G)
                       <>
                         (aeson-1.5.6.0:Data.Aeson.Types.ToJSON.pair "Speed")
                           (toJSON arg2_ay6H)) }
  toEncoding
    = \ value_ay6I
        -> case value_ay6I of {
             Vibrate arg1_ay6J arg2_ay6K
               -> aeson-1.5.6.0:Data.Aeson.Types.ToJSON.fromPairs
                    ((aeson-1.5.6.0:Data.Aeson.Types.ToJSON.pair "Index")
                       (toEncoding arg1_ay6J)
                       <>
                         (aeson-1.5.6.0:Data.Aeson.Types.ToJSON.pair "Speed")
                           (toEncoding arg2_ay6K)) }
instance FromJSON Vibrate where
  parseJSON
    = \ value_ay6L
        -> case value_ay6L of
             aeson-1.5.6.0:Data.Aeson.Types.Internal.Object recObj_ay6M
               -> ((Vibrate
                      <$>
                        ((((Data.Aeson.TH.lookupField parseJSON)
                             "Buttplug.Core.Message.Vibrate")
                            "Vibrate")
                           recObj_ay6M)
                          (Data.Text.pack "Index"))
                     <*>
                       ((((Data.Aeson.TH.lookupField parseJSON)
                            "Buttplug.Core.Message.Vibrate")
                           "Vibrate")
                          recObj_ay6M)
                         (Data.Text.pack "Speed"))
             other_ay6N
               -> (((Data.Aeson.TH.parseTypeMismatch' "Vibrate")
                      "Buttplug.Core.Message.Vibrate")
                     "Object")
                    (Data.Aeson.TH.valueConName other_ay6N)
```

**Comment by sullyj3** [comment]
this kind of boilerplate seems like potentially a good application for llms actually

**Comment by sullyj3** [comment]
closed by #14
